FAAC: unfreeRedistributable -> lgpl2

Krutonium opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Hear me out: FAAC is an MP3 encoding library from around 1997 (based on the license) according to https://github.com/knik0/faac

It used to be very patent encumbered, hence the text stating that patent royalties may be required for derivative works, despite the software itself being LGPL. That's all fine, except that in 2017 Fraunhofer, the patent holder, decided to not renew their patents, allowing them to lapse and become invalid. 

https://www.theregister.com/2017/05/16/mp3_dies_nobody_noticed/
https://www.tunequest.org/a-big-list-of-mp3-patents/20070226/

As such, I do believe that the license should be updated to reflect that new reality.

Krutonium commented:
I did very little in the way of testing as there isn't any functional changes aside from which license is in use.

ners commented:
LGTM
I've inspected the source this points to, and it indeed has the LGPL2 licence.

piegamesde commented:
Your pull request has over three thousand commits. It looks like to messed up your base branch somehow. Please rebase or cherry-pick your changes onto latest master.

Krutonium commented:
> Your pull request has over three thousand commits. It looks like to messed up your base branch somehow. Please rebase or cherry-pick your changes onto latest master.

Working on it, sorry!

Krutonium committed:
faac: Change license to lgpl2Plus

Patents making it non-redistributable expired in 2017, making it distributable.

stigtsp reviewed (changes requested):
Hi there!

Please format the PR title and commit message appropriately. There are some examples here:

https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes

jtojnar commented:
As per [contribution guidelines](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#submitting-changes), the commit message should start with the attribute that it affects. So something like:

> faac: Change license to LGPL
>
> Patents making it non-redistributable expired in 2017.

Furthermore `lgpl2` is unclear, please use `lgpl2Plus` or `lgpl2Only`. See https://discourse.nixos.org/t/lib-licenses-gpl3-co-are-now-deprecated/8206 for more details.

Krutonium commented:
>ISO/IEC gives users of the MPEG-2 NBC/MPEG-4 Audio standards free license to this software module or modifications thereof for use in hardware or software products claiming conformance to the MPEG-2 NBC/ MPEG-4 Audio standards.

Applications using the library usually advertise as such.

>Those intending to use this software module in hardware or software products are advised that this use may infringe existing patents.

This part no longer matters, as they neither control nor enforce any Patents on MP3 since 2017

>Copyright is not released for non MPEG-2 NBC/MPEG-4 Audio conforming products. 

Not relevant here

jtojnar commented:
> > Copyright is not released for non MPEG-2 NBC/MPEG-4 Audio conforming products.
> 
> Not relevant here

As I understand it, you cannot have free software with limitations like that.

Krutonium commented:
Any chance someone here is properly knowledgeable about this in terms of rule of the law? I can see both sides of this argument and I'm curious about what the norm here might be.

mweinelt commented:
Here is what debian thinks

```
Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
Upstream-Name: FAAC
Source: https://github.com/knik0/faac
Disclaimer: This package contains software that has been originally
 developed in the course of development of the MPEG-2 NBC/MPEG-4 Audio
 standard ISO/IEC 13818-7, 14496-1,2 and 3. This software module is an
 implementation of a part of one or more MPEG-2 NBC/MPEG-4 Audio tools
 as specified by the MPEG-2 NBC/MPEG-4 Audio standard.  While the
 resulting programs and binaries result in redistributable software, the
 effects on application packages derived from this library remain
 unclear. Therefore, the Debian 'non-free' section is an adequate place
 until the licenses for the problematic parts have been clarified by
 relicensing or reimplementation.
 .
 This package can legitimately and technically be auto-built, its license
 does not disapprove of automated building nor does it impose restrictions
 related to architecture or distribution.

Files: *
Copyright: © 1999-2001, Menno Bakker
           © 2002-2017, Krzysztof Nikiel
           © 2004, Dan Villiom P. Christiansen
           © 2004, Volker Fischer
License: LGPL-2.1+
 This library is free software; you can redistribute it and/or
 modify it under the terms of the GNU Lesser General Public
 License as published by the Free Software Foundation; either
 version 2.1 of the License, or (at your option) any later version.
 .
 This library is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 Lesser General Public License for more details.
 .
 You should have received a copy of the GNU Lesser General Public
 License along with this package; if not, write to the Free Software
 Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301 USA
 .
 On Debian systems, the complete text of the GNU Lesser General
 Public License can be found in `/usr/share/common-licenses/LGPL-2.1'.

Files: frontend/getopt.*
Copyright: © 1987-1993, Free Software Foundation, Inc.
License: GPL-2+

Files: frontend/mp4write.*
       libfaac/huff2.*
       libfaac/huffdata.*
       libfaac/quantize.*
       libfaac/stereo.*
Copyright: © 2017, Krzysztof Nikiel
License: GPL-3+

Files: libfaac/bitstream.*
       libfaac/channels.c
       libfaac/filtbank.c
       libfaac/tns.*
Copyright: © 1996-1997
License: other
 This software module was originally developed in the course of
 development of the MPEG-2 NBC/MPEG-4 Audio standard ISO/IEC 13818-7,
 14496-1,2 and 3. This software module is an implementation of a part of
 one or more MPEG-2 NBC/MPEG-4 Audio tools as specified by the MPEG-2
 NBC/MPEG-4 Audio standard. ISO/IEC gives users of the MPEG-2 NBC/MPEG-4
 Audio standards free license to this software module or modifications
 thereof for use in hardware or software products claiming conformance
 to the MPEG-2 NBC/ MPEG-4 Audio standards. Those intending to use this
 software module in hardware or software products are advised that this
 use may infringe existing patents. The original developer of this
 software module and his/her company, the subsequent editors and their
 companies, and ISO/IEC have no liability for use of this software
 module or modifications thereof in an implementation. Copyright is not
 released for non MPEG-2 NBC/MPEG-4 Audio conforming products. The
 original developer retains full right to use the code for his/her own
 purpose, assign or donate the code to a third party and to inhibit
 third party from using the code for non MPEG-2 NBC/MPEG-4 Audio
 conforming products. This copyright notice must be included in all
 copies or derivative works.

Files: libfaac/kiss_fft/*
Copyright: © 2003-2004, Mark Borgerding
License: BSD-3-clause
 Redistribution and use in source and binary forms, with or without
 modification, are permitted provided that the following conditions are met:
 .
 * Redistributions of source code must retain the above copyright notice, this
   list of conditions and the following disclaimer.
 * Redistributions in binary form must reproduce the above copyright notice,
   this list of conditions and the following disclaimer in the documentation
   and/or other materials provided with the distribution.
 * Neither the author nor the names of any contributors may be used to endorse
   or promote products derived from this software without specific prior written
   permission.
 .
 THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
 ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
 WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
 ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
 (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
 LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
 ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
 SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.

Files: debian/*
Copyright: © 2005-2018, Fabian Greffrath <fabian@debian.org>
License: GPL-2+

License: GPL-2+
 This package is free software; you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation; either version 2 of the License, or
 (at your option) any later version.
 .
 This package is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.
 .
 You should have received a copy of the GNU General Public License
 along with this program. If not, see <http://www.gnu.org/licenses/>
 .
 On Debian systems, the complete text of the GNU General
 Public License version 2 can be found in "/usr/share/common-licenses/GPL-2".

License: GPL-3+
 This program is free software: you can redistribute it and/or modify
 it under the terms of the GNU General Public License as published by
 the Free Software Foundation, either version 3 of the License, or
 (at your option) any later version.
 .
 This package is distributed in the hope that it will be useful,
 but WITHOUT ANY WARRANTY; without even the implied warranty of
 MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 GNU General Public License for more details.
 .
 You should have received a copy of the GNU General Public License
 along with this program. If not, see <https://www.gnu.org/licenses/>.
 .
 On Debian systems, the complete text of the GNU General
 Public License version 3 can be found in "/usr/share/common-licenses/GPL-3".
```

I think the proprietary parts do taint the package.

arianvp commented:
I agree with @jtojnar 's assessment to keep the `unfreeRedistributable` label. Especially because we're not legal experts it's probably safer to err on the safe side with these things.
